src/api/config/mod.rs

```diff
@@ -0,0 +1,61 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Route handler
+//! for the `POST /config` endpoint.
+
+#[cfg(test)]
+mod test;
+
+use rocket::{
+    data::{self, FromData},
+    http::Status,
+    Data, Request, State,
+};
+use rocket_contrib::Json;
+
+use db::config::{Config as ConfigDb, Data as ConfigData};
+use logging::MozlogLogger;
+use types::error::{AppError, AppErrorKind, AppResult};
+
+/// Payload for `POST /config`.
+#[derive(Debug, Deserialize)]
+struct Payload {
+    /// Flag indicating whether this payload clobbers existing config or should be merged with it.
+    clobber: Option<bool>,
```

Comment:
It might be a little clearer if it was called `replace` and noted that default is to merge configs.